python/ray/dashboard/client/src/pages/dashboard/logical-view/Actor.tsx

```diff
@@ -179,14 +202,97 @@ class Actor extends React.Component<Props & WithStyles<typeof styles>, State> {
             {
               label: "Required resources",
               value:
-                Object.entries(actor.requiredResources).length > 0 &&
-                Object.entries(actor.requiredResources)
-                  .sort((a, b) => a[0].localeCompare(b[0]))
-                  .map(([key, value]) => `${value.toLocaleString()} ${key}`)
-                  .join(", ")
+                Object.entries(actor.requiredResources).length > 0
+                  ? Object.entries(actor.requiredResources)
+                      .sort((a, b) => a[0].localeCompare(b[0]))
+                      .map(([key, value]) => `${value.toLocaleString()} ${key}`)
+                      .join(", ")
+                  : null
             }
           ];
 
+    // Apply transformation to add styling for information field
+    const transforms: Record<
+      string,
+      (info: ActorInformation) => ActorInformation
+    > = {
+      Pending: (info: ActorInformation) => {
+        if (actor.state !== -1 && actor.taskQueueLength >= 50) {
+          info.rendered = (
+            <React.Fragment key={info.label}>
+              <span className={classes.warningTooManyPendingTask}>
+                {info.label}: {info.value}
+              </span>
+            </React.Fragment>
+          );
+        }
+        return info;
+      },
+      ActorTitle: (info: ActorInformation) => {
+        info.rendered = (
+          <React.Fragment key={info.label}>
+            <span className={classes.actorTitle}>{info.value}</span>{" "}
+          </React.Fragment>
+        );
+        return info;
+      },
+      NumObjectIdsInScope: (info: ActorInformation) => {
+        info.rendered = (
+          <React.Fragment key={info.label}>
+            <span className={classes.secondaryFieldsHeader}>
+              <br></br>
+              {info.label}: {info.value}
+            </span>
+          </React.Fragment>
+        );
+        return info;
+      },
+      NumLocalObjects: (info: ActorInformation) => {
+        info.rendered = (
+          <React.Fragment key={info.label}>
+            <span className={classes.secondaryFields}>
+              {info.label}: {info.value}
+            </span>
+          </React.Fragment>
+        );
+        return info;
+      },
+      UsedLocalObjectMemory: (info: ActorInformation) => {
+        info.rendered = (
+          <React.Fragment key={info.label}>
+            <span className={classes.secondaryFields}>
+              {info.label}: {info.value}
+            </span>
+          </React.Fragment>
+        );
+        return info;
+      }
+    };
+    // Apply the styling transformation
+    information = information.map(val => {
+      const transform = transforms[val.label];
+      if (transform !== undefined) {
+        return transform(val);
+      } else {
+        return val;
+      }
+    });
```

Comment:
The fact that this is both mutating the values and doing a map is a little confusing and goes against the typical pure usage of `map`. Why not turn this into an immutable map, e.g. have the initial information list above have elements of type `{ label: string; value: string | null; }`, then transform it into a list with elements of type `{ label: string; node: ReactNode; }` (where a `ReactNode` is a built-in type alias that can be a JSX element, a string, null, etc.). Then the transform functions above can simply return the transformed element rather than mutating the input.